supplemental-docs/CLIENTS.md

```diff
@@ -151,6 +151,61 @@ const client = new S3Client({
 });
 ```
 
+#### Enabling uncached/refreshed credentials in `fromIni` credential provider
+
+`fromIni` credential provider accepts a boolean `ignoreCache` option which when true, always reloads credentials from the file system instead of using cached values. This is useful when you need to detect changes to the credentials file.
+
+Note: For temporary credentials that need regular refreshing, consider using `fromTemporaryCredentials` instead.
+
+Using ignoreCache with an S3 client:
+
+```typescript
+import { S3Client } from "@aws-sdk/client-s3";
+import { fromIni } from "@aws-sdk/credential-provider-ini";
+
+// Create client with credentials that will reload from file
+const client = new S3Client({
+  credentials: fromIni({ ignoreCache: true }),
+});
+```
+
+Refreshing credentials for an existing client:
+
+```typescript
+import { S3Client } from "@aws-sdk/client-s3";
+import { fromIni } from "@aws-sdk/credential-provider-ini";
+
+// Initial client setup
+const client = new S3Client({
+  credentials: fromIni({ ignoreCache: true }),
+});
+
+// To refresh credentials later:
+async function refreshClientCredentials() {
+  const credProvider = fromIni({ ignoreCache: true });
+  // Get fresh credentials and update the client
+  const freshCredentials = await credProvider();
+  client.config.credentials = freshCredentials;
```

Comment:
we shouldn't demonstrate anything that involves writing to the config object after client instantiation